### PR TITLE
Add anagram helper to crossword

### PR DESCRIPTION
**Pull request by oliverabrahams**
## What are you changing?

- Adding anagram helper

Screenshot 2024-11-29 at 12 00 02
Screenshot 2024-11-29 at 11 59 32
Screenshot 2024-11-29 at 11 59 25

### CODE REVIEW

File: libs/@guardian/react-crossword/src/components/SolutionDisplay.tsx [hunk 1]
```diff
@@ -0,0 +1,200 @@
...
+					>
+						<input
+							aria-label={`cell ${index}`}
+							ref={(el) => (inputRefs.current[index] = el)}
+							draggable={true}
+							value={candidateLetters[index]}
+							onDragStart={() => setDragItemIndex(index)}
+							onDragEnter={() => setDragOverItemIndex(index)}
+							onDragEnd={onDragEnd}
+							onChange={() => {}}
```
Comment (sndrs) [inline libs/@guardian/react-crossword/src/components/SolutionDisplay.tsx:158]:
```suggestion
							onChange={() => {}} // TODO: remove the need for this
```

File: libs/@guardian/react-crossword/src/components/SolutionDisplayCell.tsx [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+			`}
+		>
+			{progressLetter.separator === '-' && (
+				<div
+					css={css`
+						position: absolute;
+						height: 2px;
+						top: ${theme.cellSize / 2 - 0.5}px;
+						left: ${theme.cellSize - 5}px;
+						width: 7px;
```
Comment (sndrs) [inline libs/@guardian/react-crossword/src/components/SolutionDisplayCell.tsx:39]: i think this values is derived in the grid itself

File: libs/@guardian/react-crossword/src/components/SolutionDisplayCell.tsx [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+						background-color: ${theme.background};
+						z-index: 1;
+					`}
+				></div>
+			)}
+			{progressLetter.number && (
+				<div
+					css={css`
+						font-size: 0.625rem;
```
Comment (sndrs) [inline libs/@guardian/react-crossword/src/components/SolutionDisplayCell.tsx:48]: this looks different to the grid again. maybe we should abstract cell styling for resuse here and in the grid?

File: libs/@guardian/react-crossword/src/components/WordWheel.tsx [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+		.sort(() => Math.random() - 0.5);
+
+	const outerLetters = letterArray;
+	const centerLetter =
+		letterArray.length > 4 ? outerLetters.shift() : undefined;
+	const radius = 70;
+	const centerX = 100;
+	const centerY = 100;
+
+	const getPosition = (index: number): { x: number; y: number } => {
```
Comment (sndrs) [inline libs/@guardian/react-crossword/src/components/WordWheel.tsx:25]: `useCallback`?

Comment (sndrs) [reply libs/@guardian/react-crossword/src/components/WordWheel.tsx:25]: infact, does this need to be in the component itself?

**Review by sndrs — APPROVED** [review]
couple of comments, but nice one!! so much better

File: libs/@guardian/react-crossword/src/components/SolutionDisplayCell.tsx (hunk 2 above)
Comment (oliverabrahams) [reply libs/@guardian/react-crossword/src/components/SolutionDisplayCell.tsx:39]: i thought it was just cellSize - 3 to cellSize + 4

File: libs/@guardian/react-crossword/src/components/SolutionDisplayCell.tsx (hunk 3 above)
Comment (oliverabrahams) [reply libs/@guardian/react-crossword/src/components/SolutionDisplayCell.tsx]: I have copied for now. I think abstraction here might not be needed

**Pull request merged**